machine/v1beta1/types_machine.go

```diff
@@ -343,3 +343,91 @@ type MachineList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []Machine `json:"items"`
 }
+
+// NetworkConfig Config specifies an IP configuration to be applied upon machine creation.
+type NetworkConfig struct {
+	// Interfaces slice of interfaces to be configured.
+	// +kubebuilder:validation:MaxItems=1
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:Required
+	Interfaces []Interface `json:"interfaces"`
+
+	// DnsResolver defines nameservers to be applied to a network interface
+	// interfaces.
+	// +kubebuilder:validation:Required
+	DnsResolver DnsResolver `json:"dns-resolver"`
+
+	// Routes routes to be applied to a network interface
+	// interfaces.
+	// +kubebuilder:validation:Required
+	Routes Routes `json:"routes"`
+}
+
+// Interface IP configuration to be applied to a network interface.
+type Interface struct {
+	// Name interface name
+	// +kubebuilder:default=ens192
+	// +optional
+	Name string `json:"name"`
+
+	// IPV4 IP configuration for network interface
+	// +kubebuilder:validation:Required
+	IPV4 IPV4Addresses `json:"ipv4"`
```

Comment:
Should the `IPV4Addresses` contain other fields then `IPV4Address` ?
Asking since we could remove the `IPV4Addresses` and use array directly
```suggestion
	IPV4 []IPV4Address `json:"ipv4"`
```